优化 WebClient 请求超时处理，避免将超时异常转换为 500 服务器错误

**问题描述**
在使用 WebClient 发起 HTTP 请求时，当请求超时（ReadTimeoutException）会被统一包装为 WebClientRequestException，最终返回 500 Internal Server Error 给客户端。这种处理方式存在两个主要问题：

1. **不必要的错误日志**：这种超时应该属于预期的异常情况（特别是在请求外部服务时），不应该以 ERROR 级别记录，这会污染日志系统。

2. **误导用户**：前台用户看到的提示是"服务器内部错误"，但实际上是因为请求的目标服务响应超时，与 Halo 服务器本身无关。

<img width="983" height="130" alt="Image" src="https://github.com/user-attachments/assets/854ce2ca-f90d-46b4-8baf-7e0d4a6f2b31" />

**复现步骤**
可能取决于用户的网络环境。
1. 发送 GET 请求到 `/apis/api.hyperlink.halo.run/v1alpha1/link-detail` 接口
2. 参数 `url` 指向一个响应较慢的服务
3. 等待直到触发 ReadTimeout 

**日志片段**
```
2026-02-22T12:49:40.552+08:00 ERROR 7 --- [reactor-http-epoll-2] a.w.r.e.AbstractErrorWebExceptionHandler : [ea05a3a0-13994]  500 Server Error for HTTP GET "/apis/api.hyperlink.halo.run/v1alpha1/link-detail?url=https://github.com/lobehub/lobehub/issues/7154"

org.springframework.web.reactive.function.client.WebClientRequestException
    at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:137) ~[spring-webflux-6.2.15.jar:6.2.15]
Caused by: io.netty.handler.timeout.ReadTimeoutException
```

**期望行为**

1. 使用 WARN 级别记录日志，避免 ERROR 级别。
2. 错误信息应该明确告知用户"请求目标服务超时"，而非"服务器内部错误"。

**完整日志**
```
2026-02-22T12:49:40.552+08:00 ERROR 7 --- [reactor-http-epoll-2] a.w.r.e.AbstractErrorWebExceptionHandler : [ea05a3a0-13994]  500 Server Error for HTTP GET "/apis/api.hyperlink.halo.run/v1alpha1/link-detail?url=https://github.com/lobehub/lobehub/issues/7154"

org.springframework.web.reactive.function.client.WebClientRequestException
	at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:137) ~[spring-webflux-6.2.15.jar:6.2.15]
	Suppressed: The stacktrace has been enhanced by Reactor, refer to additional information below: 
Error has been observed at the following site(s):
	*__checkpoint ⇢ Request to GET https://github.com/lobehub/lobehub/issues/7154 [DefaultWebClient]
	*__checkpoint ⇢ run.halo.app.security.InitializeRedirectionWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.cache.page.PageCacheWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ SwitchUserWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ AuthorizationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ ExceptionTranslationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ LogoutWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ ServerRequestCacheWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityContextServerWebExchangeWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ AnonymousAuthenticationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.social.login.security.SocialAuthenticatorFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ top.ilay.authpasskey.PasskeyAuthenticationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.oauth.HaloOAuth2AuthenticationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ MapOAuth2AuthenticationFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ AuthenticationWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢  [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.social.login.security.SocialAuthorizationRequestRedirectWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.oauth.HaloOAuth2RedirectWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ CaptchaWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ DeviceSessionFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ ReactorContextWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ CsrfWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ CorsWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ HttpHeaderWriterWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ SecurityWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ ServerWebExchangeReactorContextWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ org.springframework.security.web.server.WebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ site.muyin.linkssubmit.filter.LinksSubmitApiWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.seo.tools.AdvancedRedirectWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.moments.rss.OldRssRouteRedirectionFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.comment.widget.captcha.CommentCaptchaFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.seo.tools.CrawlRecordFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.router.TrailingSlashRedirectFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ cc.ryanc.staticpages.endpoint.RewriteOnNotFoundFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ com.timxs.storagetoolkit.filter.ImageProcessingWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.app.infra.webfilter.AdditionalWebFilterChainProxy [DefaultWebFilterChain]
	*__checkpoint ⇢ org.springframework.web.filter.reactive.ServerWebExchangeContextFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.app.infra.webfilter.LocaleChangeWebFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ org.springframework.web.filter.reactive.UrlHandlerFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ run.halo.app.theme.UserLocaleRequestAttributeWriteFilter [DefaultWebFilterChain]
	*__checkpoint ⇢ HTTP GET "/apis/api.hyperlink.halo.run/v1alpha1/link-detail?url=https://github.com/lobehub/lobehub/issues/7154" [ExceptionHandlingWebHandler]
Original Stack Trace:
		at org.springframework.web.reactive.function.client.ExchangeFunctions$DefaultExchangeFunction.lambda$wrapException$9(ExchangeFunctions.java:137) ~[spring-webflux-6.2.15.jar:6.2.15]
		at reactor.core.publisher.MonoErrorSupplied.subscribe(MonoErrorSupplied.java:55) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.Mono.subscribe(Mono.java:4576) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxPeek$PeekSubscriber.onError(FluxPeek.java:222) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.MonoFlatMapMany$FlatMapManyMain.onError(MonoFlatMapMany.java:205) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.SerializedSubscriber.onError(SerializedSubscriber.java:124) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.whenError(FluxRetryWhen.java:229) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxRetryWhen$RetryWhenOtherSubscriber.onError(FluxRetryWhen.java:279) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onError(FluxContextWrite.java:121) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.maybeOnError(FluxConcatMapNoPrefetch.java:327) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.onNext(FluxConcatMapNoPrefetch.java:212) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.SinkManyEmitterProcessor.drain(SinkManyEmitterProcessor.java:476) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.SinkManyEmitterProcessor$EmitterInner.drainParent(SinkManyEmitterProcessor.java:620) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxPublish$PubSubInner.request(FluxPublish.java:874) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.request(FluxContextWrite.java:136) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.request(FluxConcatMapNoPrefetch.java:337) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.request(FluxContextWrite.java:136) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.Operators$DeferredSubscription.request(Operators.java:1742) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onError(FluxRetryWhen.java:196) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.core.publisher.MonoCreate$DefaultMonoSink.error(MonoCreate.java:205) ~[reactor-core-3.7.14.jar:3.7.14]
		at reactor.netty.http.client.HttpClientConnect$HttpObserver.onUncaughtException(HttpClientConnect.java:425) ~[reactor-netty-http-1.2.13.jar:1.2.13]
		at reactor.netty.ReactorNetty$CompositeConnectionObserver.onUncaughtException(ReactorNetty.java:715) ~[reactor-netty-core-1.2.13.jar:1.2.13]
		at reactor.netty.resources.DefaultPooledConnectionProvider$DisposableAcquire.onUncaughtException(DefaultPooledConnectionProvider.java:225) ~[reactor-netty-core-1.2.13.jar:1.2.13]
		at reactor.netty.resources.DefaultPooledConnectionProvider$PooledConnection.onUncaughtException(DefaultPooledConnectionProvider.java:478) ~[reactor-netty-core-1.2.13.jar:1.2.13]
		at reactor.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:245) ~[reactor-netty-core-1.2.13.jar:1.2.13]
		at reactor.netty.channel.FluxReceive.onInboundError(FluxReceive.java:466) ~[reactor-netty-core-1.2.13.jar:1.2.13]
		at reactor.netty.channel.ChannelOperations.onInboundError(ChannelOperations.java:535) ~[reactor-netty-core-1.2.13.jar:1.2.13]
		at reactor.netty.channel.ChannelOperationsHandler.exceptionCaught(ChannelOperationsHandler.java:153) ~[reactor-netty-core-1.2.13.jar:1.2.13]
		at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:346) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:325) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.AbstractChannelHandlerContext.fireExceptionCaught(AbstractChannelHandlerContext.java:317) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireExceptionCaught(CombinedChannelDuplexHandler.java:424) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.ChannelHandlerAdapter.exceptionCaught(ChannelHandlerAdapter.java:92) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.CombinedChannelDuplexHandler$1.fireExceptionCaught(CombinedChannelDuplexHandler.java:145) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.ChannelInboundHandlerAdapter.exceptionCaught(ChannelInboundHandlerAdapter.java:143) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.CombinedChannelDuplexHandler.exceptionCaught(CombinedChannelDuplexHandler.java:231) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:346) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.AbstractChannelHandlerContext.invokeExceptionCaught(AbstractChannelHandlerContext.java:325) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.AbstractChannelHandlerContext.fireExceptionCaught(AbstractChannelHandlerContext.java:317) ~[netty-transport-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.handler.timeout.ReadTimeoutHandler.readTimedOut(ReadTimeoutHandler.java:98) ~[netty-handler-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.handler.timeout.ReadTimeoutHandler.channelIdle(ReadTimeoutHandler.java:90) ~[netty-handler-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.handler.timeout.IdleStateHandler$ReaderIdleTimeoutTask.run(IdleStateHandler.java:525) ~[netty-handler-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.handler.timeout.IdleStateHandler$AbstractIdleTask.run(IdleStateHandler.java:497) ~[netty-handler-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:160) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408) ~[netty-transport-classes-epoll-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.130.Final.jar:4.1.130.Final]
		at java.base/java.lang.Thread.run(Unknown Source) ~[na:na]
Caused by: io.netty.handler.timeout.ReadTimeoutException
```

--- a/packages/hyperlink-card/src/HyperlinkCard.svelte
+++ b/packages/hyperlink-card/src/HyperlinkCard.svelte
@@ -38,7 +38,6 @@
   let siteData = $state<SiteData>();
 
   async function fetchSiteData() {
-    console.log("fetchSiteData", customTitle, customImage, customDescription);
     if (customTitle && customImage && customDescription) {
       siteData = {
         title: customTitle,
@@ -53,6 +52,10 @@
       loading = true;
 
       const response = await fetch(`/apis/api.hyperlink.halo.run/v1alpha1/link-detail?url=${href}`);
+
+      if (!response.ok) {
+        return;
+      }
 
       siteData = (await response.json()) as SiteData;
 

--- a/packages/hyperlink-card/src/HyperlinkInlineCard.svelte
+++ b/packages/hyperlink-card/src/HyperlinkInlineCard.svelte
@@ -43,6 +43,10 @@
 
       const response = await fetch(`/apis/api.hyperlink.halo.run/v1alpha1/link-detail?url=${href}`);
 
+      if (!response.ok) {
+        return;
+      }
+
       siteData = (await response.json()) as SiteData;
 
       if (customTitle) {

--- a/src/main/java/run/halo/editor/hyperlink/handler/HyperLinkBilibiliParser.java
+++ b/src/main/java/run/halo/editor/hyperlink/handler/HyperLinkBilibiliParser.java
@@ -3,12 +3,16 @@ package run.halo.editor.hyperlink.handler;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import io.netty.channel.ConnectTimeoutException;
+import io.netty.handler.timeout.ReadTimeoutException;
 import lombok.RequiredArgsConstructor;
 import org.springframework.core.io.buffer.DataBuffer;
 import org.springframework.core.io.buffer.DataBufferUtils;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.client.reactive.ReactorClientHttpConnector;
 import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientRequestException;
+import org.springframework.web.server.ServerWebInputException;
 import reactor.core.publisher.Mono;
 import run.halo.editor.hyperlink.HttpClientFactory;
 import run.halo.editor.hyperlink.dto.HyperLinkBaseDTO;
@@ -25,6 +29,14 @@ public class HyperLinkBilibiliParser implements HyperLinkParser<HyperLinkBaseDTO
 
     public Mono<HyperLinkBaseDTO> parse(URI linkURI) {
         return getHyperLinkDetail(linkURI)
+            .onErrorMap(throwable -> {
+                if (throwable instanceof WebClientRequestException wcre
+                    && (wcre.getCause() instanceof ReadTimeoutException
+                        || wcre.getCause() instanceof ConnectTimeoutException)) {
+                    return new ServerWebInputException("Request to target service timed out.");
+                }
+                return throwable;
+            })
             .map(item -> {
                 var hyperLinkDTO = new HyperLinkBaseDTO();
                 try {

--- a/src/main/java/run/halo/editor/hyperlink/handler/HyperLinkDefaultParser.java
+++ b/src/main/java/run/halo/editor/hyperlink/handler/HyperLinkDefaultParser.java
@@ -1,5 +1,7 @@
 package run.halo.editor.hyperlink.handler;
 
+import io.netty.channel.ConnectTimeoutException;
+import io.netty.handler.timeout.ReadTimeoutException;
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
@@ -18,6 +20,7 @@ import org.springframework.http.client.reactive.ReactorClientHttpConnector;
 import org.springframework.util.CollectionUtils;
 import org.springframework.util.StringUtils;
 import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientRequestException;
 import org.springframework.web.server.ServerWebInputException;
 import reactor.core.publisher.Mono;
 import run.halo.app.infra.utils.PathUtils;
@@ -37,6 +40,14 @@ public class HyperLinkDefaultParser implements HyperLinkParser<HyperLinkBaseDTO>
     @Override
     public Mono<HyperLinkBaseDTO> parse(URI linkURI) {
         return getHyperLinkDetail(linkURI)
+                .onErrorMap(throwable -> {
+                    if (throwable instanceof WebClientRequestException wcre
+                        && (wcre.getCause() instanceof ReadTimeoutException
+                            || wcre.getCause() instanceof ConnectTimeoutException)) {
+                        return new ServerWebInputException("Request to target service timed out.");
+                    }
+                    return throwable;
+                })
                 .switchIfEmpty(
                         Mono.error(new ServerWebInputException("this website is not supported.")))
                 .map(item -> {

--- a/src/main/java/run/halo/editor/hyperlink/handler/HyperLinkQQMusicParser.java
+++ b/src/main/java/run/halo/editor/hyperlink/handler/HyperLinkQQMusicParser.java
@@ -3,10 +3,14 @@ package run.halo.editor.hyperlink.handler;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import io.netty.channel.ConnectTimeoutException;
+import io.netty.handler.timeout.ReadTimeoutException;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.client.reactive.ReactorClientHttpConnector;
 import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientRequestException;
+import org.springframework.web.server.ServerWebInputException;
 import reactor.core.publisher.Mono;
 import run.halo.editor.hyperlink.HttpClientFactory;
 import run.halo.editor.hyperlink.dto.HyperLinkBaseDTO;
@@ -24,6 +28,14 @@ public class HyperLinkQQMusicParser implements HyperLinkParser<HyperLinkBaseDTO>
 
     public Mono<HyperLinkBaseDTO> parse(URI linkURI) {
         return getHyperLinkDetail(linkURI)
+            .onErrorMap(throwable -> {
+                if (throwable instanceof WebClientRequestException wcre
+                    && (wcre.getCause() instanceof ReadTimeoutException
+                        || wcre.getCause() instanceof ConnectTimeoutException)) {
+                    return new ServerWebInputException("Request to target service timed out.");
+                }
+                return throwable;
+            })
             .map(item -> {
                 var hyperLinkDTO = new HyperLinkBaseDTO();
                 try {

